### ISSUE TITLE
add support for FLOAT data type (even as an alias for DOUBLE)

### ISSUE THREAD

**Issue by gideonshalev** [trinodb/trino#7184]
FLOAT seems to be part pf the sql ansi standard (definitely a part of sql92), but is not supported by Trino.
This causes issues with some connectors, like the JDBC connector when used with a 'picky' application like Tableau.
Tableau loves the FLOAT data type, and insists on wrapping all custom sql user queries for 'optimization' purposes. When it comes to float-like data types, it uses float (not double or real). This immediately breaks any query handed off to Trino, and there doesn't seem to be a way around this.
One easy solution from the Trino perspective is to alias FLOAT to DOUBLE or REAL.

**Comment by martint** [comment]
> One easy solution from the Trino perspective is to alias FLOAT to DOUBLE or REAL.

FLOAT is a different type than DOUBLE or REAL. It's actually `FLOAT(P)`, too. From the SQL spec:

```
29) 14 FLOAT specifies the data type approximate numeric, with binary 
    precision equal to or greater than the value of the specified <precision>. 
    The maximum value of <precision> is implementation-defined. 
    <precision> shall not be greater than this value.
```

The precision is optional and implementation-defined, though, so we could make it be equivalent to DOUBLE's or REAL's precision, but the type needs to be implemented as a distinct type in the engine.

**Comment by gideonshalev** [comment]
@martint good point,I hadn't looked closely at the standard, just knew float was there. 
Would it be a big deal to get it into the engine? If not, please do so as you guys move WAAAAAAY faster than the clowns at Tableau...

**Comment by martint** [comment]
It requires some work but it's not rocket science to add it to the engine. We just need someone to volunteer to do it (cc @kasiafi? 😉 ).
